src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -500,6 +500,28 @@ public final void verifyWithGoogleConfigParser(String[] listOfModules,
         verifyViolations(filePath, violations, actualViolations);
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
+        final Configuration googleConfig = getMasterGoogleConfig();
```

Comment:
This type of method should not be in `AbstractIt`. This class is used for multiple different areas of IT, including Sun and CS's regression area. It doesn't make sense for sun to be able to see google's config. This is why the abtsracts are separated out like this. This type of mixing is almost like you gave production code access to test code. They should be kept separate.

Why can't we have:
* `verifyWithItConfig` method in this abstract class, which will take in the entire config.
* `verifyWithWholeConfig` in `AbstractGoogleModuleTestSupport` which will call `verifyWithItConfig` and pass it Google's config.

Reply:
Agree, nothing special for Google here. It just execution based on some config. Each exact test will define what config to use.

Reply:
so we will need to make changes for `verifyWithGoogleConfigParser()` too to meet these criteria.

Reply:
>Each exact test will define what config to use.

`AbstractGoogle` should define the config to use. Test should define the verify method to use.

Reply:
Yes, we can have some abstract class for all Google ITs to define path it.
Parent class for `abstract class for all Google ITs` should just receive any path to config and just load it.

Reply:
I have moved all the google related methods to the `AbstractGoogleModuleTestSupport` and kept only required methods in `AbstractItModuleTestSupport`

created a new [verifyWithItConfig()](https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L465) as @rnveach suggested.

This method takes a configuration and file path and performs the verification. We're not loading configuration in this method, we're receiving it from parameters, here is why I think it is better:

In future if we decide to migrate sun style to chapter wise testing or add new style which will follow chapter wise testing, it can work in the same way as google. We will load it's configuration from it's abstract class and pass it to this method as we're doing currently ( providing config from `AbstractGoogleModuleTestSupport` ). This will remove the overhead of loading all types of configuration from one place. We can provide the configuration of different styles from their corresponding abstract class. This is more flexible way.

Reply:
Now the `AbstractItModuleTestSupport` has only `verifyWithItConfig()` and some other helper methods to perform the verification. All the google related methods has been moved to `AbstractGoogleModuleTestSupport`